server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt

```diff
@@ -321,17 +323,19 @@ object Chapter {
         }
     }
 
-    fun getRecentChapters(): List<MangaChapterDataClass> {
-        return transaction {
-            (ChapterTable innerJoin MangaTable)
-                .select { (MangaTable.inLibrary eq true) and (ChapterTable.fetchedAt greater MangaTable.inLibraryAt) }
-                .orderBy(ChapterTable.fetchedAt to SortOrder.DESC)
-                .map {
-                    MangaChapterDataClass(
-                        MangaTable.toDataClass(it),
-                        ChapterTable.toDataClass(it)
-                    )
-                }
+    fun getRecentChapters(pageNum: Int): PaginatedList<MangaChapterDataClass> {
+        return paginatedFrom(pageNum) {
+            transaction {
+                (ChapterTable innerJoin MangaTable)
+                    .select { (MangaTable.inLibrary eq true) and (ChapterTable.fetchedAt greater MangaTable.inLibraryAt) }
+                    .orderBy(ChapterTable.fetchedAt to SortOrder.DESC)
```

Comment:
I would suggest adding this query instead of using a custom function that handles the data after the fact(though I would be fine without it)
```suggestion
                    .orderBy(ChapterTable.fetchedAt to SortOrder.DESC)
                    .limit(50, 50 * pageNum)
```

Reply:
That's what I intend to do in the following pull requests, I devised this design so we can quickly add pagination to various api endpoints.

Reply:
Got it 👍